### PR TITLE
[r] Support enumerations in `update_obs`/`update_var`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** **Issue and/or context:** Validates `update_obs`/`update_var` when the newly added column is an enumeration: tracking issue #1710, with grandparent #866.

**Changes:**

Update `SOMADataFrame$update()` to handle enumerated columns (factors)

**Notes for Reviewer:**

### CODE REVIEW

**Comment by eddelbuettel** [comment]
(The 'CI will fail for R' comment above has been take care of by merging #1717 which took of that too.)

**Comment by johnkerl** [comment]
Now with tiledb-r 0.21.1, green for review ☘️

File: apis/r/tests/testthat/test-SOMAExperiment.R [hunk 1]
```diff
@@ -80,6 +80,7 @@ test_that("Update obs and var", {
   # obs: drop an existing column and add a new one
   tbl_obs0$qux <- tbl_obs0$bar
   tbl_obs0$bar <- NULL
+  # tbl_0bs0$baz <- xxx enum
```
Comment (eddelbuettel) [inline apis/r/tests/testthat/test-SOMAExperiment.R:83]: Can be removed

File: libtiledbsoma/src/soma/column_buffer.cc [hunk 2]
```diff
@@ -116,14 +124,16 @@ ColumnBuffer::ColumnBuffer(
     size_t num_bytes,
     bool is_var,
     bool is_nullable,
-    std::optional<Enumeration> enumeration)
+    std::optional<Enumeration> enumeration,
+    bool is_ordered)
```
Comment (eddelbuettel) [inline libtiledbsoma/src/soma/column_buffer.cc:128]: This may warrant a default of 'false'.  Ordered is rare.

**Review by eddelbuettel — COMMENTED** [review]
Looks good, includes parts I contributed, ~two~ one small editorial nits inline.  (Other one was a thinko, removed.)

**Review by eddelbuettel — APPROVED** [review]

**Pull request merged**